libs/auth-api-lib/src/lib/delegations/admin/delegation-admin-custom.service.ts

```
@@ -1,108 +1,222 @@
 import { DelegationsIndexService } from '../delegations-index.service'
 import { DelegationScopeService } from '../delegation-scope.service'
-import { NoContentException } from '@island.is/nest/problem'
-import { Sequelize } from 'sequelize-typescript'
+import { CreatePaperDelegationDto } from '../dto/create-paper-delegation.dto'
+import { DelegationDTO } from '../dto/delegation.dto'
+import { NamesService } from '../names.service'
+import { DELEGATION_TAG, ZENDESK_CUSTOM_FIELDS } from '../constants/zendesk'
+import { DelegationDelegationType } from '../models/delegation-delegation-type.model'
+import { DelegationsIncomingCustomService } from '../delegations-incoming-custom.service'
+import { DelegationValidity } from '../types/delegationValidity'
 
 @Injectable()
 export class DelegationAdminCustomService {
   constructor(
     @InjectModel(Delegation)
     private delegationModel: typeof Delegation,
+    @InjectModel(DelegationDelegationType)
+    private delegationDelegationTypeModel: typeof DelegationDelegationType,
+    private readonly zendeskService: ZendeskService,
     private delegationResourceService: DelegationResourcesService,
+    private delegationsIncomingCustomService: DelegationsIncomingCustomService,
     private delegationIndexService: DelegationsIndexService,
     private delegationScopeService: DelegationScopeService,
+    private namesService: NamesService,
     private sequelize: Sequelize,
   ) {}
 
+  private getNationalIdsFromZendeskTicket(ticket: Ticket): {
+    fromReferenceId: string
+    toReferenceId: string
+  } {
+    const fromReferenceId = ticket.custom_fields.find(
+      (field) => field.id === ZENDESK_CUSTOM_FIELDS.DelegationFromReferenceId,
+    )
+    const toReferenceId = ticket.custom_fields.find(
+      (field) => field.id === ZENDESK_CUSTOM_FIELDS.DelegationToReferenceId,
+    )
+
+    if (!fromReferenceId || !toReferenceId) {
+      throw new Error('Zendesk ticket is missing required custom fields')
```

This should throw `BadRequestException` as user required input is missing.